assets/js/main.js

```diff
@@ -238,8 +238,10 @@ function onLoginLoaded(err) {
     setScalableWidth($("#overlay-container").get(0), 300);
     setScalableHeight($("#overlay-container").get(0), 450);
     $("#register-via-login-button").click(function(e) {
+        const userName = $("#input-login-username").val();
         clearOverlay(e);
         openRegister(e);
+        if (userName) $("#input-register-username").val(userName);
```

Comment:
I just tested it on macOS Firefox and it doesn't seem to work. I believe that you need to wait until the registration form is actually loaded onto the screen, then you can hook onto the element using jQuery after that. I recommend creating a callback parameter in `openRegister` function, invoke the callback if it's not undefined inside thee `req.onload` callback, then on line 243 where `openRegister` is called, pass a callback that contains this line in it.

I'm not sure why I pass `e` into `openRegister` function call, I believe some of the functions in that file operate on the target element, but that function does not appear to do anything with `e` so you can discard it and put the callback in there instead.

Reply:
Okay, I will look into it.

Reply:
@Coteh made the changes as suggested but I couldn't test as I am facing issues while setting the project on my local device.
Do let me know after checking if any further changes are required. Hope it will work now.